src/types/pointer_constraints_v1.zig

```diff
@@ -0,0 +1,80 @@
+const wlr = @import("../wlroots.zig");
+
+const pixman = @import("pixman");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+const zwp = wayland.server.zwp;
+
+pub const PointerConstraint = extern struct {
+    pub const State = extern struct {
+        pub const field = struct {
+            pub const region = 1 << 0;
+            pub const cursor_hint = 1 << 1;
+        };
```

Comment:
I've been using packed structs for bitfields like this elsewhere. See for example:
https://github.com/swaywm/zig-wlroots/blob/8a7bd64c8d727d233016333105f83f1ac706fad1/src/types/foreign_toplevel_management_v1.zig#L27-L37